pkg/clustermanager/cluster_manager.go

```diff
@@ -91,7 +91,7 @@ type ClusterClient interface {
 	CreateNamespace(ctx context.Context, kubeconfig string, namespace string) error
 	GetNamespace(ctx context.Context, kubeconfig string, namespace string) error
 	ValidateControlPlaneNodes(ctx context.Context, cluster *types.Cluster, clusterName string) error
-	ValidateWorkerNodes(ctx context.Context, clusterName string, kubeconfigFile string) error
+	ValidateWorkerNodes(ctx context.Context, clusterName string, kubeconfigFile string) (int, int, error)
```

Comment:
can you name these returns? :)
Specially with this method name, it's quite difficult to know what those ints are

Reply:
Better yet - don't return those 2 things. Of the 3 places I found this function being used, _none of them_ used either of the returned integers. The 1 place I found they _might_ be used but its unclear is l.715 of those file. I'm not sure how that `policy` inline funciton definition is being used but its awfully convoluted. 

At a high level, looking at these changes, there's a mixture of 'Validate' and 'Retrieve' behavior munged into single functions that really needs to be pulled apart to honor the SRP.